schema/shared/src/test/scala-2/zio/blocks/schema/DerivedOpticsSpec.scala

```diff
@@ -0,0 +1,369 @@
+package zio.blocks.schema
+
+import scala.language.reflectiveCalls
+import zio.test._
+
+object DerivedOpticsSpec extends ZIOSpecDefault {
+
+  // ===== Test Types =====
+
+  // Basic case class
+  final case class Person(name: String, age: Int)
+  object Person extends DerivedOptics[Person] {
+    implicit val schema: Schema[Person] = Schema.derived
+  }
+
+  // Nested case class
+  final case class Address(street: String, city: String)
+  object Address extends DerivedOptics[Address] {
+    implicit val schema: Schema[Address] = Schema.derived
+  }
+
+  final case class Employee(name: String, address: Address)
+  object Employee extends DerivedOptics[Employee] {
+    implicit val schema: Schema[Employee] = Schema.derived
+  }
+
+  // Single field case class
```

Comment:
ZIO Schema also supports "newtypes", which are seen in Scala 3 as opaque types, and should be tested. 

For example:

```scala
opaque type Age = Int
object Age {
  def apply(i: Int): Age = i 
}
```

The derived schema will be a `Wrapper` (Reflect.Wrapper), and should result in a `Lens[Age, Int]` (I think; actually would be better as `Iso[Age, Int]`, but we don't have `Iso` yet in optics).